Add note on using non-default ALSA PCM devices to bluealsa-aplay.1.rst (#571)
Arkq/bluez-alsa: doc/bluealsa-aplay.1.rst

    Select ALSA playback PCM device to use for audio output.
    The default is ``default``.

    Available ALSA PCM devices can be listed by running **aplay -L**. You must

>     use the full PCM device name eg `front:CARD=Generic_1,DEV=0` not just `front`.

Reviewer: Suggest "To use any other device than the default audio card, you must use the full device name as listed eg 'plughw:CARD=Generic_1,DEV=0'.

Reviewer 2: but ... does "plughw:..." appear in the `aplay -L` output for you? it does not for me. I think that there is no "simple" way to explain ALSA PCM configuration to inexperienced users in one or two lines. That, I am sure, is why no other ALSA application attempts to do it. If there is a reliable, comprehensive, beginners' guide out there then we could refer users to it. But I have not yet found one that does not raise more questions than it answers. So my gut feeling is that we should avoid any kind of ALSA tutorial, no matter how simple, within the BlueALSA manual pages. Wikis and Gists is where that kind of material belongs, IMHO. But of course, these are just the ramblings of an old man, what do I know?

Reviewer 3: I've checked my 3 Linux distros and the output is not consistent :D

|Gentoo | Ubuntu 20.04 | Debian 11|
|- | - | -|
|no `plughw:` | `plughw:CARD=PCH,...` | `plughw:CARD=vc4hdmi0,...`
|no `hw:` | `hw:CARD=PCH,...` | `hw:CARD=vc4hdmi0,...`
|`front:CARD=PCH,...` | `front:CARD=PCH,...` | no `front:`

I kinda do agree with @borine , manual pages in bluez-alsa are not ALSA guides.... But also one more line will not hurt anyone unless it will create more doubts. In that case it's better to link proper ALSA tutorial.

Reviewer 2: The output from `aplay -L` is actually the list of all PCMs ***that have a namehint with `description` defined and `show` set to true***. So it depends on various defaults that my be overridden by some distributions, config nodes created by various additional packages, and config nodes created or overridden by admins and users. There may be other useable PCMs that are not shown in the list simply because they do not have such namehints.

There just is no straightforward answer to this. Putting in nothing (for example `man aplay` says for its `-D` option:
``` 
 -D, --device=NAME
              Select PCM by name
```
is clearly meaningless to anyone using ALSA for the first time. But on the other hand, putting in a statement that is only correct in some cases but incomplete or wrong in others can make thinks worse by discouraging users from searching elsewhere for help. If only there was a "proper" ALSA tutorial, but I don't know of one.

Reviewer 3: > actually the list of all PCMs that have a namehint with description defined and show set to true

Yeap... I have forgotten about that... :D So, even stating that `aplay -L` is the list of PCMs might mislead someone :D But somehow I feel that we *can* add one statement about the `aplay -L` as a hint, but it should be worded differently.

@danboid so that you understand the scale of the problem here. When you run `aplay -L` you probably do not see PCM device named `bluealsa`... but after installing bluez-alsa, there should be such :D And that's one of reasons why we've got PulseAudio and PipeWire :D For a normal Linux user, ALSA is too difficult to use/configure.

Author: Yes, ALSA is a mess and hence PA, Pipewire etc.

Apparently NetBSD uses the same audio system that Solaris used. I wonder what it might be missing that I need?

Anyone here ever used the OpenBSD audio system, I've forgotten its name? It has be better than ALSA.

>     Available ALSA PCM devices can be listed by running **aplay -L**. You must
>     use the full PCM device name eg `front:CARD=Generic_1,DEV=0` not just `front`.

Reviewer 3: OK, so I will use the power of repo owner and will settle this :D (unless someone will say "veto")

```suggestion
    On most Linux distributions, available ALSA PCM devices can be listed by running **aplay -L**.
```

Put it at the end of the `--pcm` help message (after "bluealsa-aplay does not perform..." paragraph; for most users this paragraph will be more important than the `aplay` hint).
Also, please add `aplay(1)` to the "SEE ALSO" list.

If you are worried that this simple statement might not be enough, add "EXAMPLES" section just before "SEE ALSO" (to be honest I'd be glad if you could provide some examples). In that section you can provide real life usage of `bluealsa-aplay` (2-3 examples) where you can use `-D` option with e.g. `front:CARD=USB,DEV=0`.
